apps/vx-scan/backend/src/app.ts

```diff
@@ -310,6 +310,9 @@ function buildApi(
     }): Promise<Result<void, Error>> {
       const authStatus = auth.getAuthStatus();
 
+      // Though we only initiate this action when a poll worker is logged in, a poll worker could
+      // remove their card right after, so we treat these as handled errors rather than unexpected
+      // errors
```

Comment:
What should happen overall in the case that a poll worker removes their card right after? Currently, we're just swallowing the error on the frontend. Will that leave things in a consistent state? The user won't necessarily know something went wrong.

This makes me think we should either use assertions here and treat them as unhandled errors (which will at least fail loudly) or think a little more about how to make this safe.

Reply:
Totally agree that we should have clear feedback for the user in this case!

In practice, if you remove your card right after this is action is triggered, the failure won't happen here but in the actual card writing logic, which also wraps errors in a `Result`. So modifying these to be assertions won't make much of practical difference. My manual testing verified this.

On that note, this is a pre-existing problem with pre-existing TODOs, e.g.

https://github.com/votingworks/vxsuite/blob/449583919a1d2f4d3b10e0c05bac7192d38b4b3b/apps/vx-scan/frontend/src/screens/poll_worker_screen.tsx#L305-L310


Since this isn't a regression, I'm going to move forward with this PR and have added this to my [miscellaneous cleanup issue](https://github.com/votingworks/vxsuite/issues/2965)!

Reply:
Also, for completeness, I tested what actually happens when you remove a card preemptively both before and after this PR, and in both cases, polls successfully transition, but the report written to the card is corrupted/unreadable

Reply:
Thanks for looking deeper into this! Given what you found out about the report data getting corrupted, it makes me think that ideally the polls transition + report data writing should be somehow transactional. Hard to get right, so maybe not important to prioritize. But I agree with you that showing an error to the user seems like a bare minimum